deployment/common/changeset/state.go

```diff
@@ -136,7 +138,23 @@ func MaybeLoadMCMSWithTimelockChainState(chain deployment.Chain, addresses map[s
 				return nil, err
 			}
 			state.CancellerMcm = mcms
+		case tvStr.Type == multichain.Type && tvStr.Version.String() == multichain.Version.String():
+			// the same contract can have different roles so we use the labels to determine which role it is
+			mcms, err := owner_helpers.NewManyChainMultiSig(common.HexToAddress(address), chain.Client)
+			if err != nil {
+				return nil, err
+			}
+			if tvStr.Labels.Contains(types.ProposerRole.String()) {
```

Comment:
Oh, you're using one entry, but with any number of labels, so they're semantically meaningful tags, not a qualifier to the address, yeah?  Totally reasonable. Just making sure I understand.